bw2analyzer/lci.py

```diff
@@ -1,14 +1,60 @@
 import pandas as pd
-from bw2data import get_activity
+from bw2data import Database
 from bw2calc import LCA
+from typing import List, Union
 
 
-def get_labeled_inventory(lca: LCA) -> pd.DataFrame:
+def _get_dependent_database_names(lca: LCA) -> List[str]:
+    databases = []
+    for key in lca.remapping_dicts.keys():
+        databases += [key[0] for key in lca.remapping_dicts[key].values()]
+    return pd.unique(databases)
+
+
+def _load_database_metadata(
+    lca: LCA, cols: Union[None, List[str], str] = None
+) -> pd.DataFrame:
+    # get database names
+    databases = _get_dependent_database_names(lca)
+    # load metadata and store in one dataframe
+    df = pd.DataFrame()
+    for db in databases:
+        df = df.append(pd.DataFrame(Database(db)), ignore_index=True)
```

Comment:
It's faster to do this:

```python
def get_db(db):
       keys = ['activity', 'activity type', 'authors', 'classifications', 'code',
               'comment', 'database', 'filename', 'flow', 'id', 'location', 'name',
               'parameters', 'production amount', 'reference product', 'synonyms',
               'type', 'unit']
       return pd.DataFrame([{key: obj.get(key) for key in keys} for obj in Database(db)])
```

2.7 versus 3.8 seconds. However, your approach is easier and more flexible, so fine with me.

Reply:
That's interesting. I thought iterating over the database was slow (see #15) but it seems another part was. Perhaps the `get_activity`?

I am trying to avoid pre-determining the keys because users might have custom keys in their databases (e.g. Exiobase uses "compartments" where Ecoinvent uses "categories"). Is it still faster if we ask for `obj.keys()` instead of the fixed list?

Reply:
It's fine, your question is the exact reason we shouldn't try to hard code things, you never know what users are up to...